### PR TITLE
Add kotlinc checker for kotlin

### PR DESCRIPTION
**Pull request by francis36012**
Hi @w0rp, finally got around to fixing my branch.

### CODE REVIEW

File: ale_linters/kotlin/kotlinc.vim [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+let g:ale_kotlin_kotlinc_options = get(g:, 'ale_kotlin_kotlinc_options', '')
+let g:ale_kotlin_kotlinc_enable_config = get(g:, 'ale_kotlin_kotlinc_enable_config', 0)
+let g:ale_kotlin_kotlinc_config_file = get(g:, 'ale_kotlin_kotlinc_config_file', '.ale_kotlinc_config')
+let g:ale_kotlin_kotlinc_classpath = get(g:, 'ale_kotlin_kotlinc_classpath', '')
+let g:ale_kotlin_kotlinc_sourcepath = get(g:, 'ale_kotlin_kotlinc_sourcepath', '')
+let g:ale_kotlin_kotlinc_use_module_file = get(g:, 'ale_kotlin_kotlinc_use_module_file', 0)
+let g:ale_kotlin_kotlinc_module_filename = get(g:, 'ale_kotlin_kotlinc_module_filename', 'module.xml')
+
+function! ale_linters#kotlin#kotlinc#GetCommand(buffer) abort
+    let l:kotlinc_opts = g:ale_kotlin_kotlinc_options
```
Comment (w0rp) [inline ale_linters/kotlin/kotlinc.vim]: In between your last pull request and now, support has been added for buffer-local variables. You'll need to change every line which reads a `g:ale_foo` variable to `ale#Var(a:buffer, 'foo')`. Have a look at the other linters.

File: ale_linters/kotlin/kotlinc.vim [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        endif
+
+        let l:type = l:match[1]
+        let l:text = l:match[2]
+
+        let l:type_marker_str = l:type ==# 'warning' || l:type ==# 'info' ? 'W' : 'E'
+
+        call add(l:output, {
+        \   'bufnr': 0,
+        \   'lnum': -1,
```
Comment (w0rp) [inline ale_linters/kotlin/kotlinc.vim]: You can just remove `lnum` and `col` here. You can also remove every use of `vcol`, `bufnr`, and `nr` now. Those are set automatically now.

**Comment by francis36012** [comment]
Thanks for feedback @w0rp. The suggested changes have been made and pushed to the branch. Can you look at it and tell me if that is enough.

**Comment by w0rp** [comment]
Cheers! :beers:

Thanks for coming back and updating this. Hopefully it will be useful for others.

**Pull request merged**